Hunting Queries/Microsoft 365 Defender/Command and Control/SuspiciousNetworkConnection-NotionAPI.yaml

```diff
@@ -0,0 +1,49 @@
+id: fe4f4336-a2e7-4cae-af43-3285316475be
+name: Suspicious network connection to Notion API - Possible Notion C2
+description: |
+  This query will look for connections made to the Notion cloud service API which are not from a browser or legitimate Notion application. The query was inspired by the Offensive Notion project: https://github.com/mttaggart/OffensiveNotion OffensiveNotion is a full-featured C2 platform built on the Notion note taking app.
+requiredDataConnectors:
+  - connectorId: MicrosoftThreatProtection
+    dataTypes:
+    - DeviceNetworkEvents
+tactics:
+  - Command and control
+technique:
+  - T1102
+query: |
+  let excludedProcesses = datatable(name:string)["browser1.exe","browser2.exe","legitAppName.exe"]; //examples but check your environment first to remove false positives and use the filename and file path to reduce risk of false negative or evasion from the bad guys
+  DeviceNetworkEvents
+  | where RemoteUrl has "api.notion.com" and InitiatingProcessVersionInfoCompanyName != "Notion Labs, Inc"
+  // not (InitiatingProcessFileName has_any (excludedProcesses)) // uncomment if you want to add exclusions to your hunt query
+  | sort by TimeGenerated asc
+  | extend HashAlgorithm = "SHA256"
+  | extend Account_0_Name = InitiatingProcessAccountName
+  | extend Host_0_HostName = DeviceName
+  | extend File_0_Name = InitiatingProcessFileName
+  | extend FileHash_0_Name = InitiatingProcessSHA256
+  | extend FileHash_0_Algorithm = HashAlgorithm
+  | extend IP_0_Address = RemoteIP
```

Comment:
Mismatch in assignment here. RemoteIP is mentioned here but CallerIpAddress is mentoined in line 48.